proof-defns/proof_defns.tex

```diff
@@ -0,0 +1,438 @@
+    The symbols $+$, $-$, and $\times$ represent the usual mathematical operations of addition, subtraction, and multiplication, respectively.
+\end{definition}
+
+\begin{definition}[$\oplus, \ominus, \otimes$]
+    The symbols $\oplus$, $\ominus$, and $\otimes$ denote the corresponding floating point operations in \href{https://www.mpfr.org/}{MPFR} / in Rust arithmetic. 
+\end{definition}
+
+\connor{Maybe change ``floating point operations in \href{https://www.mpfr.org/}{MPFR}'' to some other thing that it denotes. I'm not sure if they should ``denote the corresponding floating point operations in \href{https://www.mpfr.org/}{MPFR}'', but I wanted to preserve the notation, and I wasn't sure what it should be defined on \ldots.}
+    
+\silvia{Use same notation in both pseudocode and proofs?}
+
+\todo{Change to cast.}\connor{I don't understand this todo; could someone else complete it?}
+\begin{definition}
+    Given a Rust value $x$, the function \texttt{val(x)} returns the actual (real or integer) number represented by the Rust value $x$.
+
+\end{definition}
+
+\horizline
+
+\subsection{Notes, todos, questions}
+
+\todo{Add the cast definition?}
+
+\todo{max, min, val(x) (give number represented by Rust value), see email from June 21. Subindex for type?}
+
+\todo{Add special float symbol for max, min?} No, from discussion on June 28: as long as we pass as arguments things that have used float symbols, it's OK.
+
+\questionc{Type in subindex vs casting type.}
+\connor{What does ``subindex for type'' mean (see todo above for use of this phrase; meaning possibly relates to question in this line)?} \silvia{I think it means deciding between, for example, $\texttt{max}_{\texttt{u32}}(x, y)$ or $\texttt{cast(max(x, y), u32)}$.}
+
+\section{Data Representation}
+
+\begin{definition}[Vector]
+A vector $v$ is an ordered list of objects.
+\end{definition}
+
+\begin{definition}[Set]
+A set is an unordered list of objects.
+\end{definition}
+
+\begin{definition}[Multiset]
+A multiset is a modification of the notion of a set which, unlike a set, allows for repetitions for each of its elements. The number of repetitions of an element in the multiset corresponds to the \textit{multiplicity} of that element. \questionc{multiset vs MultiSet.} \silvia{Add what we said on July 15 (Connor).}
+\end{definition}
+
+%\silvia{Sets are by definition unordered, no need to specify that in the 2nd sentence.}
+
+\begin{remark}
+Given a vector $v$, we denote its multiset representation by $\MultiSet(v)$. This distinction is relevant because vectors are ordered objects whereas multisets are not. In the OpenDP library, all datasets are represented as vector domains, and therefore for any vector $v$ we need to use the notation $\MultiSet(v)$ when referring to its multiset representation to indicate that ordering should be dropped. \questionc{Clashes a bit with the $\MultiSet(\X)$ use in the PF; domain vs vector.} \silvia{This should be resolved after consensus on July 15.}
```

Comment:
Universal quantifiers like "all" are hairy. Not all datasets are represented as vector domains. Dataframes are datasets that have a MapDomain. Potentially StringDatasets as well, which have a Carrier type of String, but a notion of symmetric distance based on carriage returns.